### ISSUE TITLE
fails to vendor dependencies

### ISSUE THREAD

**Issue by ognevny** [zed-industries/zed#10460]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

run `cargo vendor --locked --versioned-dirs` in the root directory of project. there will be an error as `tracing 0.1.40` has 2 sources

### Environment

this is the build issue, so
`rustc -Vv`:
```
rustc 1.77.2 (25ef9e3d8 2024-04-09) (Rev1, Built by MSYS2 project)
binary: rustc
commit-hash: 25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04
commit-date: 2024-04-09
host: x86_64-pc-windows-gnu
release: 1.77.2
LLVM version: 18.1.3
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

```
error: failed to sync

Caused by:
  found duplicate version of package `tracing v0.1.40` vendored from two sources:

        source 1: registry `crates-io`
        source 2: https://github.com/tokio-rs/tracing?rev=tracing-subscriber-0.3.18#8b7a1dde
```

**Comment by ognevny** [comment]
commit made that change: https://github.com/zed-industries/zed/commit/f6c85b28d5c69d9231fd10b2808ce18a34a96647

**Comment by ognevny** [comment]
fixed at f2fc84ab44d70a4f60d497e3d8cf3f1e862560cf
